File: ios/flutter_naver_map/Sources/flutter_naver_map/util/location/NDefaultMyLocationTrackerPermissionStatus.swift
import CoreLocation

enum NDefaultMyLocationTrackerPermissionStatus: String {
    case granted = "granted"
    case denied = "denied"
    case deniedForever = "deniedForever"
    
    func toMessageable() -> String {
        return self.rawValue
    }
}

extension CLAuthorizationStatus {
    func toNDefaultMyLocationTrackerPermissionStatus() -> NDefaultMyLocationTrackerPermissionStatus? {
        switch self {
        case .authorizedAlways, .authorizedWhenInUse:
            return .granted
        case .denied:
            return .denied
        case .restricted:
            return .deniedForever
        case .notDetermined:
            return nil
        }
    }
    
    func toMessageable() -> String? {
        self.toNDefaultMyLocationTrackerPermissionStatus()?.toMessageable()
    }
}

File: android/src/main/kotlin/dev/note11/flutter_naver_map/flutter_naver_map/util/location/NDefaultMyLocationTrackerPermissionStatus.kt
package dev.note11.flutter_naver_map.flutter_naver_map.util.location

enum class NDefaultMyLocationTrackerPermissionStatus(val value: String) {
    Granted("granted"),
    Denied("denied"),
    DeniedForever("deniedForever");

    fun toMessageable(): String {
        return value
    }
}